test_fms/random_numbers/test_random_numbers.F90

```diff
@@ -0,0 +1,279 @@
+program test_random_numbers
+
+use fms_mod, only: fms_init, fms_end
+use platform_mod, only: r4_kind, r8_kind
+use mpp_mod, only: mpp_error, fatal, stderr, mpp_npes, mpp_pe
+use fms_string_utils_mod, only: string
+use time_manager_mod, only: time_type, get_date, set_date, set_calendar_type, JULIAN
+use random_numbers_mod
+
+implicit none
+
+integer, parameter :: n_moments = 1000 !> Highest-order moment to test
+integer, parameter :: n0_1d = 2500 !> Initial length of 1D random sample vector
+integer, parameter :: n0_2d = 50 !> Initial dimensions of 2D random sample array
+
+integer, parameter :: seeds(0:*) = [0, -5, 3] !> Seed constants
```

Comment:
You're a Fortran guru, this is the first time I'm seeing something like this lol.  Does this notation work with all compilers?

Reply:
I've tested it with Intel, GNU, and Cray and it works with all of them. It seems that assumed-size arrays are actually older than the dinosaurs (they're a Fortran 77 feature), while square bracket array constructors are an F2003 feature.

Reply:
can you confirm that this works with the nvdia compilers?

Reply:
Changed to a 1-indexed array in b978a99d766f774876429aff0d0d0143e070fcf6.